### PR TITLE
Added node version check

### PR DESCRIPTION
**Pull request by arich-gavant**
Added a `checkNodeVersion` middleware check to verify the process version on execution and exits if invalid.

(Should `checkNodeVersion` be in a `/middleware` directory or `/utils`? It felt not worth it for the one function, and if I could think of another check we would implement I'd have gone ahead and done it. But it felt a bit overkill for this.)

Updated `help` to `demandCommand` (which I think was outdated? I saw the same in the docs and both were given as valid answers but only `demandCommand` worked).

Screenshot 2023-10-17 at 10 38 47 AM

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -8,8 +8,20 @@ const runReactAppBlueprint = require('../blueprints/react-app');
 
 const usage = chalk.cyan(`\nHi, I'm Gavin! 👋\nUsage: gavin [command] [options]`);
 
+// TODO: Would be nice to have a bundler handle syncing the volta version with this one
+const nodeMajor = 18;
```
Comment (billdami) [inline bin/index.js:12]:
```suggestion
const pkg = require('../package.json');
const nodeMajor = Number(pkg.volta.node.split('.')[0]);
```

This should work, but definitely test it out

File: bin/index.js [hunk 2]
```diff
@@ -8,8 +8,20 @@ const runReactAppBlueprint = require('../blueprints/react-app');
...
+
+const checkNodeVersion = () => {
+    if (parseInt(process.version.split('.')?.[0]?.replace('v', '')) < nodeMajor) {
+        console.log(chalk.redBright('Command failed'))
+        console.log(chalk.redBright(`Gavin formally requests you upgrade to Node ${nodeMajor}+`))
+        process.exit(1)
+    }
+}
```
Comment (billdami) [inline bin/index.js:20]: lets move this out to `../utils/node.js` if we can. trying to keep the bin/index as lean as possible, as basically just the yargs entrypoint/defintion file.

File: bin/index.js [hunk 3]
```diff
@@ -45,5 +57,5 @@ yargs(hideBin(process.argv))
         type: 'boolean',
         description: 'Run with verbose logging',
     })
-    .help()
+    .demandCommand()
```
[the remark this answers is not in the document]
Comment (arich-gavant) [reply bin/index.js:60]: I don't believe it was doing anything, unless the thing it was doing is not what I thought it was doing?

I was expecting, when typing `gavin` with no input, to get the `gavin help` prompt back. But nothing happened. Whereas `demandCommand` yields that output

File: bin/index.js (hunk 1 above)
Comment (arich-gavant) [reply bin/index.js:12]: Screenshot 2023-10-17 at 11 00 54 AM

:nice:

**Review by billdami — APPROVED** [review]

**Pull request merged**